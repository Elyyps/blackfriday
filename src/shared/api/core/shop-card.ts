import { TimeLeftBar } from "./time-left-bar";
import { Link } from "./link";

export class ShopCard {
  public brand: string;
  public button: Link;
  public category: string;
  public content: string;
  public picture: string;
  public postedAt: string;

  public seeMore: Link;
  public timeLeftBar: TimeLeftBar;
  public title: string;
}
export const generateDummyShopCards = (): ShopCard[] => {
  const data: ShopCard[] = [];
  for (let i = 0; i < 200; i++) {
    data.push({
      button: { title: "button", url: "" },
      content: "content",
      picture: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/1.png",
      seeMore: { title: "button", url: "" },
      timeLeftBar: { text: "", value: 5 },
      title: "title " + i,
      postedAt: "27/05/1995",
      category: "fdsfds",
      brand: "coolblue"
    });
  }

  return data;
};
export function generateData(): ShopCard[] {
  const data = [];
  const brand = ["jbl", "adidas", "guess", "converse", "only"];
  const category = ["audio", "sport", "mode", "games", "electronic"];
  const status = ["Actie nog onbekend", "Actie start over 5 dagen!", "Nu geldig", "Bijna Verlopen!", "Actie verlopen"];
  for (let y = 0; y < 5; y++) {
    for (let i = 1; i <= 5; i++) {
      const number16 = Math.floor(Math.random() * (1 - 0)) + 0;

      const timebar = Math.random();

      data.push({
        button: { title: "Naar deals", url: "" },
        content: "content" + i,
        picture: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/" + i + ".png",
        seeMore: { title: "Meer info", url: "" },
        timeLeftBar: { text: status[y], value: timebar },
        title: "Item n: " + i,
        postedAt: "7/06/2019",
        category: category[y],
        brand: brand[y]
      });
    }
  }

  return data;
}
