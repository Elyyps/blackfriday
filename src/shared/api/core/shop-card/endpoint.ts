import {
  generateData,
  generateDummyShopCards,
  generateEmptyDummyShopCards,
  generateDemoDummyData
} from "./generate-dummy-data";

export const generateShopCardData = generateData;
export const generateShopCardData2 = generateDummyShopCards;
export const generateEmptyShopCardData = generateEmptyDummyShopCards;
export const generateDemoData = generateDemoDummyData;
