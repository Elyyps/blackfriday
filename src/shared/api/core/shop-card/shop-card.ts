import { TimeLeftBar } from "../time-left-bar";
import { Link } from "../link";

export class ShopCard {
  public brand: string;
  public button: Link;
  public category: string;
  public content: string;
  public picture: string;
  public seeMore: Link;
  public timeLeftBar: TimeLeftBar;
  public title: string;
}
