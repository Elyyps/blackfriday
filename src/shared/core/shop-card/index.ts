export * from './shop-card.component'