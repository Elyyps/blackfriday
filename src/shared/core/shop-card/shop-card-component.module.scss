.shop-card {
  position: relative;
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  padding: 16px;
  background: $global-inverse-color;
  position: relative;
  width: 100%;
  height: 288px;
  border-radius: 2px;
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.08);
}

.label {
  position: absolute;
  z-index: 10;
  left: 8px;
  top: 8px;
  display: flex;
  flex-direction: row;
  align-items: center;
  background: $brand-primary;
  border-radius: 2px;
  color: $global-inverse-color;
  font-weight: 700;
  font-size: 11px;
  text-transform: uppercase;
  padding: 4px 8px;
  line-height: 11px;
  span:first-child {
    margin-right: 4px;
  }
}

.image {
  width: 100%;
  height: 90px;
  order: 1;
  a {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    img {
      -ms-flex: 1;
      max-width: 60%;
      max-height: 60%;
    }
  }
}

.status {
  width: 100%;
  order: 2;
  &-title {
    font-size: 11px;
    text-transform: uppercase;
    font-weight: 700;
    color: $gray-light;
    &.green {
      color: $brand-secondary;
    }
    &.orange {
      color: #ff9d0a;
    }
  }
}

.content {
  width: 100%;
  order: 3;
  &-title h4 {
    margin-bottom: 4px;
  }
  &-description {
    color: $gray-light;
    margin-right: 4px;
    font-size: 14px;
    width: 100%;
    line-height: 18px;
  }
  &-link {
    a {
      font-size: 14px !important;
      @media (max-width: 959px) {
        width: auto !important;
      }
    }
  }
}
.action {
  order: 4;
  margin-top: auto;
  &,
  a {
    width: 100%;
  }
}

@mixin responsive-card-768 {
  &,
  .shop-card {
    height: 156px;
    flex-direction: row;
    flex-wrap: wrap;
  }

  .label {
    left: auto;
    right: 8px;
  }

  .content {
    margin-right: 16px;
    order: 1;
    flex: 1 1 60%;
  }

  .image {
    order: 2;
    flex: 1 1 30%;
    height: 72px;
    a {
      justify-content: flex-end;
    }
  }

  .status {
    order: 3;
    flex: 1 1 30%;
    margin-top: auto;
    margin-right: 16px;
    margin-bottom: 10px;
    &-bar {
      display: none;
    }
  }

  .action {
    order: 4;
    flex: 1 1 10%;
    justify-content: flex-end;
  }
}

@mixin responsive-card-640 {
  .content {
    flex: 1 1 80%;
  }
  .image {
    flex: 1 1 10%;
    height: 72px;
    a img {
      width: auto;
      max-width: 100%;
      max-height: 60%;
    }
  }
}

@mixin responsive-card-460 {
  .image {
    flex: 1 1 20%;
  }
  .content {
    flex: 1 1 70%;
  }
  .action {
    order: 4;
    flex: 1 1 52%;
    justify-content: flex-end;
    margin-top: 16px;
  }
  .status {
    flex: 1 1 38%;
  }
}

@media (max-width: 768px) {
  @include responsive-card-768();
}

@media (max-width: 640px) {
  @include responsive-card-640();
}

@media (max-width: 460px) {
  @include responsive-card-460();
}

.responsive {
  @include responsive-card-768();
  .action {
    flex: 1 1 40%;
  }

  @media (max-width: 768px) {
    .status {
      flex: 1 1 30%;
    }
    .action {
      flex: 1 1 10%;
    }
  }

  @media (max-width: 640px) {
    @include responsive-card-640();
    .status {
      flex: 1 1 30%;
    }
    .action {
      flex: 1 1 40%;
    }
  }

  @media (max-width: 460px) {
    @include responsive-card-460();
    .status {
      flex: 1 1 10%;
    }
    .action {
      flex: 1 1 50%;
    }
  }

  @media (min-width: 769px) {
    .action {
      flex: 1 1 60%;
    }
  }

  .content {
    flex: 1 1 70%;
  }
  .image {
    flex: 1 1 20%;
    height: 72px;
    a img {
      width: auto;
      max-width: 100%;
      max-height: 60%;
    }
  }
}
