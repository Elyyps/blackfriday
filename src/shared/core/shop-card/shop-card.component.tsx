
import * as React from "react";
import styles from "./shop-card-component.module.scss";

export interface IShopCardComponentProps {

}

const ShopCardComponent = (props: IShopCardComponentProps) => {
  return ( <div className={styles["shop-card"]}>

  </div>)
};

export { ShopCardComponent };
  