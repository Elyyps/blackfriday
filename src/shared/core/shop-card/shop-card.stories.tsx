
import React from "react";
import { shallow } from "enzyme";
import { ShopCardComponent } from "./shop-card.component";

describe("[ShopCard]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <ShopCardComponent />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});

  