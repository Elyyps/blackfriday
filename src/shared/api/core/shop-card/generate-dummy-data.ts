import { ShopCard } from "./shop-card";
import { generateFilterBarData } from "../filter-bar/endpoint";

export const generateDummyShopCards = (): ShopCard[] => {
  const data: ShopCard[] = [];
  const numberOfCard = 100;
  for (let i = 0; i < numberOfCard; i += 1) {
    data.push({
      button: { title: "button", url: "" },
      content: "content",
      picture: "https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/1.png",
      seeMore: { title: "button", url: "" },
      timeLeftBar: { text: "", value: 5 },
      title: "title ",
      category: "audio",
      brand: "coolblue"
    });
  }

  return data;
};
export const generateEmptyDummyShopCards = (): ShopCard[] => [
  {
    button: { title: "", url: "" },
    content: "",
    picture: "",
    seeMore: { title: "", url: "" },
    timeLeftBar: { text: "", value: 5 },
    title: " ",

    category: "",
    brand: ""
  }
];

export const generateData = (): ShopCard[] => {
  const data = [];
  const fiveConst = 5;
  const brand = ["jbl", "adidas", "guess", "converse", "only"];
  const category = ["audio", "sport", "mode", "games", "electronic"];
  const status = ["Actie nog onbekend", "Actie start over 5 dagen!", "Nu geldig", "Bijna Verlopen!", "Actie verlopen"];
  for (let y = 0; y < fiveConst; y += 1) {
    for (let i = 1; i <= fiveConst; i += 1) {
      const timebar = Math.random();

      data.push({
        button: { title: "Naar deals", url: "" },
        content: "content",
        picture: `https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/${i}.png`,
        seeMore: { title: "Meer info", url: "" },
        timeLeftBar: { text: status[y], value: timebar },
        title: "Item n: ",
        category: category[y],
        brand: brand[y]
      });
    }
  }

  return data;
};
export const generateDemoDummyData = (): ShopCard[] => {
  const data: ShopCard[] = [];
  const brand = ["Appel", "Adidas", "Converse", "Guess", "JBL", "Only", "Test1", "Test2", "Test3"];
  const status = ["Actie nog onbekend", "Actie start over 5 dagen!", "Nu geldig", "Bijna Verlopen!", "Actie verlopen"];
  const content = [
    "Sint aanbiedingen: -30% op heel veel mooie merken!",
    "Een selectie van bizarre vele aanbiedingen.",
    "Een selectie van bizarre vele aanbiedingen. Meer info",
    "Een selectie van bizarre vele aanbiedingen. Meer info",
    "Een selectie van bizarre vele aanbiedingen"
  ];
  const filterBarData = generateFilterBarData();
  const fiveConst = 5;
  for (let a = 0; a < filterBarData.categories.length; a += 1) {
    for (let b = 0; b < filterBarData.categories[a].label; b += 1) {
      const image = Math.floor(Math.random() * (fiveConst - 0)) + 0;
      const brands = Math.floor(Math.random() * (fiveConst - 0)) + 0;

      data.push({
        button: { title: "Naar deals", url: "" },
        content: content[brands],
        picture: `https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/shopCards/${image}.png`,
        seeMore: { title: "Meer info", url: "" },
        timeLeftBar: { text: status[brands], value: brands / fiveConst },
        title: brand[brands],
        category: filterBarData.categories[a].text,
        brand: brand[brands]
      });
    }
  }

  return data;
};
