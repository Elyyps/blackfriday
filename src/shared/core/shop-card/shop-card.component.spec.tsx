import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
//import { withProvider } from "@app/modules/core/storybook-provider";
import { ShopCardComponent } from "./shop-card.component";

storiesOf("ShopCard", module).addDecorator(withA11y);
//.addDecorator(withProvider)
//.add("Basic implementation", () => (
//  <ShopCardComponent />
//));
