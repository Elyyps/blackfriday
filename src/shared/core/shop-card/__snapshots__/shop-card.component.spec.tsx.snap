// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ShopCard] should render component without crashing 1`] = `
<ShadowCardComponent
  shadowSize="small"
>
  <div
    className="shop-card"
  >
    <div
      className="shop-card__body"
    >
      <div
        className="shop-card__image"
      >
        <ImageComponent />
      </div>
      <div
        className="shop-card__status-title none"
      />
      <TimeLeftBarComponent
        color="none"
        range={0}
      />
      <div
        className="shop-card__content"
      >
        <div
          className="shop-card__title"
        />
         
      </div>
    </div>
    <div
      className="shop-card__action"
    >
      <Button
        icon="link.svg"
        iconPosition="right"
      />
    </div>
  </div>
</ShadowCardComponent>
`;
